From 6defc0175194f6bf016643ba470e406193c1985d Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 11 Dec 2019 13:36:59 +0100
Subject: [PATCH 1/1] mkimage: simplify fit_check_params()

fit_check_params() wants at least two of dflag, fflag, and lflag set.

Simplify the logical constraint checking this.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 tools/fit_image.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/fit_image.c b/tools/fit_image.c
index 0201cc44d8..c204d58fb6 100644
--- a/tools/fit_image.c
+++ b/tools/fit_image.c
@@ -820,9 +820,9 @@ static int fit_check_params(struct image_tool_params *params)
 {
 	if (params->auto_its)
 		return 0;
-	return	((params->dflag && (params->fflag || params->lflag)) ||
-		(params->fflag && (params->dflag || params->lflag)) ||
-		(params->lflag && (params->dflag || params->fflag)));
+	return	((params->dflag && params->fflag) ||
+		 (params->fflag && params->lflag) ||
+		 (params->lflag && params->dflag));
 }
 
 U_BOOT_IMAGE_TYPE(
-- 
2.24.0

